### PR TITLE
PMM-7674 Add allocated and available disk

### PR DESCRIPTION
**Pull request by tiagomotasantos**
**What this PR does / why we need it**: Adds allocated and available disk size to DBaaS cluster creation. Also refactors the resources values to handle different units (currently only using it for disk since is the one that can have the biggest possible values for the available field).

**Which issue(s) this PR fixes**: https://jira.percona.com/browse/PMM-7674

Screenshot 2021-03-16 at 21 49 07

### CODE REVIEW

File: public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts [hunk 1]
```diff
@@ -82,9 +82,28 @@ export interface DBClusterExpectedResources {
...
+export interface ResourcesWithUnits {
+  value: number;
+  units: ResourcesUnits | CpuUnits;
+}
+
+export enum ResourcesUnits {
+  BYTES = 'Bytes',
+  KB = 'KB',
+  MB = 'MB',
+  GB = 'GB',
```
Comment (fabio-silva) [inline public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts:99]: Quick question, maybe @atymchuk can answer it: is there a reason why we're using GB at some places, like here, and GiB in others like Backups?

**Review by fabio-silva — APPROVED** [review]

File: public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts (hunk 1 above)
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts:99]: If you mean the label maybe is just inconsistency (or not). `GiB` is used when you represent the value with powers of 2, `GB` is used when you represent the value with powers of 10. In DBaaS we are using powers of 10 (for simplicity and because Kubernetes handles it), hence the "GB".

Comment (atymchuk) [reply public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts:100]: IMHO, we should be using GiB everywhere, hence base 2. We are in IT, so 2 makes more sense :)

Comment (browseman) [reply public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts:100]: I would suggest [this](https://en.wikipedia.org/wiki/Gigabyte) read, it explain well the difference between the GB and GiB. 

TL;DR; - GiB make sense to be used for describing RAM or any other device that it's physical design is tight to power of 2. For example, a good use is if you want to be 100% precise when printing a value (1 GiB vs 1.074 GB) and don't want to look pedantic in the same place. At the end the amount that is being described is the same, just presented differently.

Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts:100]: @atymchuk k8s speaks whatever we want based on the notation we pass (see [here](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory)). We are using base 10, hence the `GB`, so cannot change the label here without changing the rest of the implementation (to avoid inconsistencies). But I agree that base 2 makes more sense for this things (e.g. memory as @browseman said), we just started with base 10 and went with it, it works fine, but definitely can be changed.

Comment (atymchuk) [reply public/app/percona/dbaas/components/DBCluster/DBCluster.types.ts:100]: Wow, I learned something. This is exactly why I love code reviews 🥇

**Comment by browseman** [comment]
A few things that I can give as feedback from the screenshot:

- Ordering on most of the platforms that I encountered is CPU, RAM, HDD. Rising this just for consideration and I'm aware that the above ordering (RAM,CPU,HDD) could be on purpose.
- `3.07GB (59.3%) of 5.18GB used` at least for me it's a bit hard to read. Again just suggesting for consideration something like: `Using 3.07GB (59.3%) of 5.18GB in total` could be easier to comprehend.
- IMO the presence of  `Required {CPU,Memory,Disk}` on the resource usage bar graph is making the reading it harder. At least from my point of view showing only the `Used/Total` is enough. 
- The presence of the red warning when there are not enough resources available is really great. 
- Also probably it will be clearer if we are a explicit and write it as `Minimum Required {CPU,Memory,Disk}` 
Please consider the above suggestions as a single point of view.

**Comment by tiagomotasantos** [comment]
Thanks a lot for the comments @browseman. Updated some of the texts to match your suggestions, I agree too. Just some notes:
- The required part was a requested featured, discussed with the team and @rnovikovP, because what we request as resources it's not what we get because of other components required for the cluster (e.g. you request 2GB of memory but actually get 4GB used due to extra components);
- The required is not "minimum required", the value shown is actually the required, meaning it shouldn't take more than that (should be exactly that, no more, no less).

Thanks a lot for the input, will also consider to change the order of the resources!!

**Review by lunaticusgreen — APPROVED** [review]
👍
